### PR TITLE
bit improved Makefile: once recursive instead of thrice

### PR DESCRIPTION
**Pull request by abdulbadii**
Help will really print Help, not process dependency

### CODE REVIEW

**Comment by snicolet** [comment]
Hi @abdulbadii 

Thanks for your pull request :-)

Could you give us a little bit more feed back about the changes you propose? For instance, what is exactly the problem you want to solve, and what technical solutions you have chosen -- a comparison of the two outputs before and after the patch would be nice too.By the way, I see that you replace = by :=, is that important or purely personal preference? these kinds of things :-)

**Comment by abdulbadii** [comment]
I too just learnt that a directive on `make` (or any interpreter) is always processed the first, just like preprocessor on a compiler, and we know that help is intended just to print out helpful message.
Hence so no matter what, if a user means it / Help, the include directive is straightly processed which has nothing to do with his need. This is meant to give more efficiency for his use by the directive checking previous its execution, that is call .depend/$(DEP) target.
If he is for Help it'll simply be skipped, else, the dependency is done as normally as the 'original makefile'
Diff is surely supported by Github/Git system which gives us a clearly comparison, we can all check it out, I of course can't afford it by myself 
:= in lieu of = are all in GNU make official standard reference/manual https://www.gnu.org/software/make/manual/make.pdf
:= immediate
= deferred
just make safer code base in case of expansion/refactoring

**Comment by abdulbadii** [comment]
@snicolet help me consider it, it's been a week withou care..

**Comment by vondele** [comment]
I'll take a look, but will have to refresh my make-foo. So, if anybody wants to do an intermediate review that would be helpful.

**Comment by abdulbadii** [comment]
@snicolet help me consider it, it's been a week withou care..

**Comment by abdulbadii** [comment]
@snicolet how is this ?
for a week being ignored or belittled

**Comment by abdulbadii** [comment]
@vondele ?

**Comment by vondele** [comment]
CI seems to fail on OSX, related to a 32/64 bit issue, if I understand correctly.

**Comment by abdulbadii** [comment]
did test On Mingw-Windows and Linux to work perfectly, all in the latest build tools

**Comment by snicolet** [comment]
Sorry, but this pull request tries to change too many things at the same time in the Makefile. 
As it stands, it opens a can of problems instead of solving anything broken, IMO.

File: src/Makefile [hunk 1]
```diff
@@ -155,7 +155,7 @@ endif
 
 ### 3.1 Selecting compiler (default = gcc)
 F :=
-CXXFLAGS += -Wall -Wcast-qual -fno-exceptions -std=c++11 $(EXTRACXXFLAGS) $(F)
+CXXFLAGS += -Wall -Wcast-qual -fno-exceptions -std=c++11 $(EXTRACXXFLAGS)$(F)
```
Comment (vondele) [inline src/Makefile:160]: can you explain what that approach with `$(F)`

Comment (abdulbadii) [reply src/Makefile:160]: apologize, indeed I'm in disaster for 3 days; my Windows crashed and so did my Linux and still try to make Iso bootable usb luckily a family member keep generous giving me online inet free to google a solution.
Just an optional flag for compiler

- simply ignore it, then just nothing
- if one opt to add it e.g.
  make build ARCH=x86-64-modern F=-flto
  and he is on Mingw64 system in that he's sure it fully works in building, executing it, then it'll build with link time optimization
- I personally find useful to ensure metaprogramming correctness by check F=-Werror=vla
 so there'll be no "fake" builtin array if it compiles perfectly

Comment (vondele) [reply src/Makefile:160]: can't we do this already with `CXXFLAGS='-Werror=vla' make -j ARCH=.....` ?

Comment (abdulbadii) [reply src/Makefile]: you're correct, the same, is simpler so easier, faster to user/cpu

**Comment by gvreuls** [comment]
The changes look okay, we have no variable reassignments so using simple assignment shouldn't pose a problem. I do think the removal of the `objclean` and `profileclean` targets and re-implementing them elsewhere to gain a little speed is overkill though.

Caveat: I can't test builds on exotic platforms, just intel linux/windows.

**Comment by vondele** [comment]
While some of the change look good, the combination of changes made it a bit difficult to review.
However, the Makefile has changed much recently, I'll close this.

**Pull request closed without merging**